swagger.json

```diff
@@ -88,95 +97,274 @@
           }
         } ],
         "responses" : {
-          "200" : {
-            "description" : "OK"
+          "201" : {
+            "description" : "TACO File binary and metadata created & processing started.",
+            "schema" : {
+              "$ref": "#/definitions/ResourceResponse"
+            }
           },
-          "405" : {
-            "description" : "Invalid input"
+          "401" : {
+            "description" : "You are not authorized to create a File in TACO."
+          },
+          "415" : {
+            "description" : "Unsupported file type provided."
+          },
+          "500" : {
+            "description" : "This File could be created at this time by TACO."
           }
         }
       }
+    },
+    "/status/{ID}" : {
+      "get" : {
+        "summary" : "Resource Processing Status.",
+        "description" : "Get the processing status and history for a resource.",
+        "operationId" : "getProcessStatus",
+        "produces" : [ "application/json" ],
+        "parameters" : [ {
+          "name" : "ID",
+          "in" : "path",
+          "description" : "SDR Identifier for the Resource.",
+          "required" : true,
+          "type" : "string"
+        } ],
+        "responses" : {
+          "200" : {
+            "description" : "Processing status for the TACO resource.",
+            "schema" : {
+              "$ref": "#/definitions/ProcessResponse"
+            }
+          },
+          "401" : {
+            "description" : "You are not authorized to view this resource's processing status in TACO."
+          },
+          "404" : {
+            "description" : "Check your identifier - resource not found."
+          },
+          "500" : {
+            "description" : "This resource's processing status could be retrieved at this time by TACO."
+          }
+      }
     }
   },
   "definitions" : {
     "Resource" : {
       "type" : "object",
-      "required" : [ "sourceId", "title" ],
+      "required" : [ "@context", "@type", "access", "label", "preserve", "publish" ],
       "properties" : {
+        "@context" : {
+          "type" : "string",
+          "format": "uri",
+          "description" : "URI for the JSON-LD context definitions",
+          "example": "http://sdr.sul.stanford.edu/contexts/taco-base.jsonld"
+        },
+        "@type" : {
+          "type" : "string",
+          "format": "uri",
+          "description": "URI for the resource type",
+          "example" : "http://sdr.sul.stanford.edu/models/sdr3-object.jsonld"
+        },
+        "access" : {
+          "type" : "string",
+          "description": "What groups should be able to access the resource in Access environments"
+        },
+        "contains" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string",
+            "format" : "uri"
+          },
+          "description": "The child resource(s) of this resource."
+        },
+        "contained-by" : {
```

Comment:
Do we need to support both relationships?  Supporting both simultaneously can make it more difficult to achieve consistency guarantees.

Reply:
yes.

Reply:
I understand the concern @jcoyne, but if we're explicit about these relationships now, we should be able to stay on top of any difficulty/confusion.